File: src/download_data.py
# author: Tiffany Timbers (Code modified by Gabe Fairbrother)
# date: 2021-11-19

"""Downloads data csv data from the web to a local filepath as a csv 

Usage: src/down_data.py --url=<url> --out_file=<out_file>

Options:
--url=<url>              URL from where to download the data (must be in standard csv format)
--out_file=<out_file>    Path (including filename) of where to locally write the file
"""
  
from docopt import docopt
import requests
import os
import pandas as pd


opt = docopt(__doc__)

def main(url, out_file):
  try: 
    request = requests.get(url)
    request.status_code == 200
  except Exception as req:
    print("Website at the provided url does not exist.")
    print(req)
    
  data = pd.read_csv(url, header=None)
  
  try:
      data.to_csv(out_file, index = False)
  except:
    os.makedirs(os.path.dirname(out_file))
    data.to_csv(out_file, index = False)
  
if __name__ == "__main__":
  main(opt["--url"], opt["--out_file"])

File: src/download_data.r
# author: Wanying Ye
# date: 2021-11-23

"Downloads delimited data from the web to a local filepath as a csv.

Usage: download_data.r --url=<url> --out_file=<out_file> [--delimiter=delimiter]

Options: 
--url=<url>               URL from where to download the data (must standard csv format)
--out_file=<out_file>     Path (including the filename) of where to write the csv file locally
[--delimiter=<delimiter>] Optional nonstandard delimiter if file has a delimiter other than ','
" -> doc

library(tidyverse)
library(readr)
library(docopt)

opt <- docopt(doc)

main <- function(url, out_file, delimiter = ',') {
    # url = "https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-red.csv"
    # url = "https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-white.csv"
    # out_file = "data/winequality-red.csv"
    # out_file = "data/winequality-white.csv"

    # read in data
    data <- read_delim(url, delim = ',')

    # write the csv file to a local path (but the output file is colon-seperated, not sure why)
    write_csv(data, out_file)
}

main(opt$url, opt$out_file, opt$delimiter)